packages/synapse-react-client/src/components/AccessRequirementList/ManagedACTAccessRequirementRequestFlow/DataAccessRequestAccessorsFilesForm.tsx

```diff
@@ -145,21 +157,128 @@ export default function DataAccessRequestAccessorsFilesForm(
+  const {
+    mutateAsync: updateRequestAsync,
+    isLoading: isLoadingUpdateDataAccessRequest,
+  } = useUpdateDataAccessRequest({
     onError: onError,
   })
 
+  const isLoading =
+    isLoadingGetDataAccessRequest ||
+    isLoadingUpdateDataAccessRequest ||
+    isLoadingSubmitDataAccessRequest
+
+  /**
+   * This effect comprises a collection of updates we should immediately apply to a data access request.
+   */
+  useEffect(() => {
+    if (dataAccessRequest && user) {
+      let shouldUpdate = false
+
+      // Attach the researchProjectId to the request
+      if (!dataAccessRequest.researchProjectId) {
+        dataAccessRequest.researchProjectId = researchProjectId
+        shouldUpdate = true
+      }
+
+      // Add the current user with GAIN_ACCESS to the list of accessors
+      const currentUserWithGainAccess: AccessorChange = {
+        userId: user.ownerId,
+        type: isRenewal ? AccessType.RENEW_ACCESS : AccessType.GAIN_ACCESS,
+      }
+      if (
+        !dataAccessRequest.accessorChanges ||
+        !dataAccessRequest.accessorChanges.find(item =>
+          deepEquals(item, currentUserWithGainAccess),
+        )
+      ) {
+        dataAccessRequest.accessorChanges = [
+          currentUserWithGainAccess,
+          ...(dataAccessRequest.accessorChanges || []),
+        ]
+        shouldUpdate = true
+      }
+
+      // SWC-5765: Filter out duplicate accessors
+      const seen = new Set()
+      const uniqueAccessorChanges = dataAccessRequest.accessorChanges.filter(
+        accessorChange => {
+          return seen.has(accessorChange.userId)
+            ? false
+            : seen.add(accessorChange.userId)
+        },
+      )
+      if (
+        uniqueAccessorChanges.length !==
+        dataAccessRequest.accessorChanges.length
+      ) {
+        dataAccessRequest.accessorChanges = uniqueAccessorChanges
+        shouldUpdate = true
+      }
+
+      if (shouldUpdate) {
+        updateRequestAsync(dataAccessRequest)
+      }
+    }
+  }, [
+    dataAccessRequest,
+    isRenewal,
+    researchProjectId,
+    updateRequestAsync,
+    user,
+  ])
+
+  /**
+   * Fields other than uploaded files are NOT immediately synced with the server. Set local state if these values exist in the
+   * fetched object, and no value exists in local state.
+   */
+  useEffect(() => {
+    if (dataAccessRequest) {
+      if (dataAccessRequest.accessorChanges && accessorChanges == undefined) {
+        setAccessorChanges(dataAccessRequest.accessorChanges)
+      }
+      if (
+        isRenewal &&
+        dataAccessRequest.publication &&
+        publication == undefined
+      ) {
+        setPublication(dataAccessRequest.publication)
+      }
+      if (
+        isRenewal &&
+        dataAccessRequest.summaryOfUse &&
+        summaryOfUse == undefined
+      ) {
+        setSummaryOfUse(dataAccessRequest.summaryOfUse)
+      }
+    }
+  }, [dataAccessRequest])
```

Comment:
If the fetched object includes data (e.g. if it was saved before submitting, or if it was a renewal), set local state. Only set local state if it doesn't exist.